src/classLibrary/Utility/GlobFactory.ts

```diff
@@ -0,0 +1,94 @@
+import path from 'path';
+import _glob from 'glob';
+import { DirectorySystemInstance, FileSystemInstance, 
+    GlobSafe, 
+    GlobSyncOptions, 
+    IFileFactory, 
+    IGlobFactory, 
+    VirtualFileSystemInstance} from "../interfaces/interfaces";
+import { FileFactory } from "./FileFactory";
+import { Bar, Presets } from 'cli-progress';
+import { Utility } from './Utility';
+import { isBinaryFile } from 'isbinaryfile';
+
+class GlobFactory implements IGlobFactory {
+    private readonly fileFactory: IFileFactory;
+    private readonly util: Utility;
+    
+    constructor(fileFactory: FileFactory, util: Utility) {
+        this.fileFactory = fileFactory
+        this.util = util;
+    }
+
+    /**
+	 * Generates the empty file according to the root source path, root destination path, glob and specified target
+	 * returns the files
+	 * @param glob Glob from CyanObject from cyan.script.js
+     * @param targetDirFromDestRoot Optional filepath where the contents of the glob root should be generated at the destination root filepath
+	 */
+    GenerateFiles(glob: GlobSafe, targetDirFromDestRoot: string = './'): VirtualFileSystemInstance[] {
+        let pattern = path.resolve(this.fileFactory.FromRoot, glob.root, glob.pattern.join(''));
```

Comment:
@kirinnee Correct me if i'm wrong here

I think `glob.pattern` as an array means that the Glob may have multiple patterns to match different sets of files.  So I don't think a `join('')` will work here.